data/group-schema/src/main/resources/net/corda/schema/group/policy.1.0/corda.group.policy.json

```diff
@@ -0,0 +1,272 @@
+      "type": "object",
+      "additionalProperties": {
+        "type": "string"
+      }
+    },
+    "p2pParameters": {
+      "description": "All parameters required to configure the P2P communication before registration and synchronisation.",
+      "type": "object",
+      "required": [
+        "sessionPki",
+        "tlsTrustRoots",
+        "tlsPki",
+        "tlsVersion",
+        "protocolMode"
+      ],
+      "properties": {
+        "sessionTrustRoots": {
+          "description": "The trust roots for session initiation certificate verification.",
+          "type": "array",
+          "items": {
+            "$ref": "#/$defs/certificate"
+          }
+        },
+        "sessionPki": {
+          "description": "The PKI type to use for session handling.",
+          "enum": [
+            "Standard",
+            "StandardEV3",
+            "Corda4",
+            "NoPKI"
+          ]
+        },
+        "tlsTrustRoots": {
+          "description": "The trust roots for TLS certificate verification.",
+          "type": "array",
+          "items": {
+            "$ref": "#/$defs/certificate"
+          }
+        },
+        "tlsPki": {
+          "description": "The PKI type to use for TLS handling.",
+          "enum": [
+            "Standard",
+            "StandardEV3",
+            "Corda4"
+          ]
+        },
+        "tlsVersion": {
+          "description": "The supported TLS version.",
+          "enum": [
+            "1.2",
+            "1.3"
+          ]
+        },
+        "protocolMode": {
+          "description": "The type of end to end session to use.",
+          "enum": [
+            "Authentication",
+            "Authentication_Encryption"
+          ]
+        }
+      },
+      "additionalProperties": false
+    },
+    "protocolParameters": {
+      "description": "All parameters required to perform the registration or synchronisation protocols.",
+      "type": "object",
+      "required": [
+        "sessionKeyPolicy"
+      ],
+      "properties": {
+        "sessionKeyPolicy": {
+          "description": "Whether to use combined or distinct keys for ledger signing and end-to-end session authentication",
+          "enum": [
+            "Combined",
+            "Distinct"
+          ]
+        },
+        "staticNetwork": {
+          "description": "The configuration for set up of a static network.",
+          "type": "object",
+          "required": [
+            "members"
+          ],
+          "properties": {
+            "members": {
+              "description": "The member templates for the members to be added to the static network member list.",
+              "type": "array",
+              "items": {
+                "description": "A member template as a map of values for a member to be added to the static network member list.",
+                "type": "object"
```

Comment:
I did have that originally, but since we changed to the key-value approach, similar to the member info `LayeredPropertyMap`, I felt it was better to keep this open for any string-string pairs since we will basically just output all of the MGMs member info here which we may add to. Also it would be hard to validate for specific keys, specifically for arrays of data where the keys will contain indexes.

Reply:
ah ok, that makes sense - I wasn't considering the free-form key-value approach will be used here too.

Reply:
Ah, sorry. My comment was in regards to MGM Info. I misread where in the schema this comment was. But the answer is essentially the same. We may have indexed values. We should aim to have good documentation supporting this.